application-services/custom/camera-management/appcamera/evam.go

```diff
@@ -218,6 +221,87 @@ func (app *CameraManagementApp) getPipelineStatus(deviceName string) (interface{
 	return nil, nil
 }
 
+// processEdgeXEvent is the function that is called when an EdgeX event is received
+func (app *CameraManagementApp) processEdgeXEvent(_ interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
+	if data == nil {
+		app.lc.Errorf("processEdgeXEvent: was called without any data")
+		return false, nil
+	}
+
+	if app.config.AppCustom.DefaultPipelineName == "" || app.config.AppCustom.DefaultPipelineVersion == "" {
+		app.lc.Errorf("processEdgeXEvent: PipelineName or PipelineVersion is empty")
+		return false, nil
+	}
+
+	systemEvent, ok := data.(dtos.SystemEvent)
+	if !ok {
+		app.lc.Errorf("type received is not a SystemEvent")
+		return false, nil
+	}
+
+	if systemEvent.Type != "device" {
+		app.lc.Debug("system event type is not device")
+		return false, nil
+	}
+
+	if systemEvent.Action != "added" {
+		app.lc.Debug("system event action is not added")
+		return false, nil
+	}
+
+	device := dtos.Device{}
+	err := systemEvent.DecodeDetails(&device)
+	if err != nil {
+		app.lc.Errorf("failed to decode device details: %v", err)
+		return false, nil
+	}
+
+	return app.startDefaultPipeline(device)
+}
+
+func (app *CameraManagementApp) startDefaultPipeline(device dtos.Device) (bool, error) {
+	pipelineRunning := app.isPipelineRunning(device.Name)
+
+	if pipelineRunning {
+		app.lc.Debugf("pipeline is already running for device %s", device.Name)
+		return false, nil
+	}
+
+	startPipelineRequest := StartPipelineRequest{
+		PipelineName:    app.config.AppCustom.DefaultPipelineName,
+		PipelineVersion: app.config.AppCustom.DefaultPipelineVersion,
+	}
+
+	protocol, ok := device.Protocols["Onvif"]
+	if ok {
+
+		app.lc.Debugf("Onvif protocol information found for device: %s message: %v", device.Name, protocol)
+
+		profileResponse, err := app.getProfiles(device.Name)
+		if err != nil {
+			app.lc.Errorf("failed to get profiles for device %s, message: %v", device.Name, err)
+			return false, err
+		}
+
+		app.lc.Debugf("Onvif profile information found for device: %s message: %v", device.Name, profileResponse)
+
+		startPipelineRequest.Onvif = &OnvifPipelineConfig{
+			ProfileToken: string(profileResponse.Profiles[0].Token),
+		}
+	}
+	if _, ok := device.Protocols["USB"]; ok {
```

Comment:
this should probably be an `else if` if possible since it cant be both usb and onvif